python/lsst/daf/base/propertyContainer/propertySet.cc

```diff
@@ -1,69 +1,74 @@
+#include "pybind11/stl.h"
+
 #include <string>
 #include <typeinfo>
 
-#include <pybind11/pybind11.h>
-#include <pybind11/stl.h>
-#include <pybind11/operators.h>
-
 #include "lsst/daf/base/Citizen.h"
 #include "lsst/daf/base/PropertySet.h"
 #include "lsst/daf/base/DateTime.h"
 
-using namespace lsst::daf::base;
-
 namespace py = pybind11;
+using namespace pybind11::literals;
 
-template <typename T, typename C> void addAccessors(C &cls, std::string const& name) {
+namespace lsst {
+namespace daf {
+namespace base {
+namespace {
+
+template <typename T, typename C> void declareAccessors(C &cls, std::string const& name) {
     const std::string getName = "get" + name;
     cls.def(getName.c_str(), (T (PropertySet::*)(std::string const&) const) &PropertySet::get<T>,
-            py::arg("name"));
+            "name"_a);
     cls.def(getName.c_str(), (T (PropertySet::*)(std::string const&, T const&) const) &PropertySet::get<T>,
-            py::arg("name"), py::arg("defaultValue"));
+            "name"_a, "defaultValue"_a);
 
     const std::string getArrayName = "getArray" + name;
     cls.def(getArrayName.c_str(), (std::vector<T> (PropertySet::*)(std::string const&) const) &PropertySet::getArray<T>,
-            py::arg("name"));
+            "name"_a);
 
     const std::string setName = "set" + name;
     cls.def(setName.c_str(), (void (PropertySet::*)(std::string const&, T const&)) &PropertySet::set<T>,
-            py::arg("name"), py::arg("value"));
+            "name"_a, "value"_a);
     cls.def(setName.c_str(), (void (PropertySet::*)(std::string const&, std::vector<T> const&)) &PropertySet::set<T>,
-            py::arg("name"), py::arg("value"));
+            "name"_a, "value"_a);
 
     const std::string addName = "add" + name;
     cls.def(addName.c_str(), (void (PropertySet::*)(std::string const&, T const&)) &PropertySet::add<T>,
-            py::arg("name"), py::arg("value"));
+            "name"_a, "value"_a);
     cls.def(addName.c_str(), (void (PropertySet::*)(std::string const&, std::vector<T> const&)) &PropertySet::add<T>,
-            py::arg("name"), py::arg("value"));
+            "name"_a, "value"_a);
 
     const std::string typeName = "TYPE_" + name;
     cls.attr(typeName.c_str()) = py::cast(typeid(T), py::return_value_policy::reference);
 }
 
-PYBIND11_DECLARE_HOLDER_TYPE(MyType, std::shared_ptr<MyType>);
+} // <anonymous>
+
+PYBIND11_PLUGIN(propertySet) {
+    py::module::import("lsst.daf.base.persistable");
 
-PYBIND11_PLUGIN(_propertySet) {
-    py::module mod("_propertySet", "Access to the classes from the daf_base propertySet library");
+    py::module mod("propertySet");
 
     py::class_<std::type_info>(mod, "TypeInfo")
-        .def(py::self == py::self)
-        .def(py::self != py::self);
+        .def("__eq__", [](std::type_info const & self, std::type_info const & other) { return self == other; })
+        .def("__ne__", [](std::type_info const & self, std::type_info const & other) { return self != other; });
```

Comment:
We should think about getting this pushed upstream somehow.  If we ever import another `pybind11` module that also wraps `std::type_info`, bad thing happen.